c7n/resources/cloudfront.py

```diff
@@ -436,3 +438,100 @@ def process_distribution(self, client, distribution):
+
+        policies:
+        - name: enforce-distribution-logging
+            resource: distribution
+            filters:
+            - type: value
+                key: "Logging.Enabled"
+                value: null
+            actions:
+            - type: update-distribution
+                update:
+                Comment: ""
+                Enabled: true
+                Logging:
+                    Enabled: true
+                    IncludeCookies: false
+                    Bucket: 'test-enable-logging-c7n.s3.amazonaws.com'
+                    Prefix: ''
+    """
+    schema = type_schema('update-distribution',
+                        update={"type": "object"},
+                        required=('update',))
+
+    permissions = ("cloudfront:UpdateDistribution",
+                   "cloudfront:GetDistributionConfig",)
+    shape = 'UpdateDistributionRequest'
+
+    def validate(self):
+        if not self.data.get('update'):
+            raise PolicyValidationError('update parameters are missing')
+        attrs = dict(self.data.get('update'))
+        if attrs.get('CallerReference') or attrs.get('Origins') or attrs.get('DefaultCacheBehavior'):
+            raise PolicyValidationError('Invalid attribute parameter')
+
+        # Set default values for required fields
+        attrs["CallerReference"] = ""
+        attrs["Origins"] = {
+            "Quantity": 0,
+            "Items": [
+                {
+                    "Id": "",
+                    "DomainName": ""
+                }
+            ],
+        }
+        attrs["DefaultCacheBehavior"] = {
+            "TargetOriginId": "",
+            "ForwardedValues": {
+                "QueryString": True,
+                "Cookies": {
+                    "Forward": ""
+                }
+            },
+            "TrustedSigners": {
+                "Enabled": True,
+                "Quantity": 0,
+            },
+            "ViewerProtocolPolicy": "",
+            "MinTTL": 0
+        }
+        request = {
+            "DistributionConfig": attrs,
+            "Id": "sample_id",
+            "IfMatch": "sample_string",
+        }
+        return shape_validate(request, self.shape, 'cloudfront')
+
+    def process(self, distributions):
+        client = local_session(self.manager.session_factory).client(
+            self.manager.get_model().service)
+        for d in distributions:
+            self.process_distribution(client, d)
+
+    def process_distribution(self, client, distribution):
+        try:
+            res = client.get_distribution_config(
+                Id=distribution[self.manager.get_model().id])
+            config = res['DistributionConfig']
+            config = {**config, **self.data['update']}
+            res = client.update_distribution(
+                Id=distribution[self.manager.get_model().id],
+                IfMatch=res['ETag'],
+                DistributionConfig=config
+            )
+        except Exception as e:
```

Comment:
might be worth continuing for resource not found exception

Reply:
this should explicitly catch a distribution not found, and raise for all other errors